Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2895,6 +2901,189 @@ namespace OpenInfraPlatform
+						std::vector<typename IfcEntityTypesT::IfcReal> coefficientZ = polynomialCurve->CoefficientsZ;
+						//convert to double
+						for (int i = 0; i < size(coefficientZ); i++)
+						{
+							polynomialConstantZ.push_back(double(coefficientZ[i]));
+						}
+						polynomialConstantCntZ = std::size(polynomialConstantZ);
+						//calculate angle
+						angleZ = SpiralUtils::AngleByAngleDeviationPolynomial(polynomialConstantZ, polynomialConstantCntZ, parameter);
+					}
+					
+					// calculate angle between two polynomial curves
+					//double angle = std::atan2(angleY, angleX);
+
+					if (polynomialConstantCntX>0 && polynomialConstantCntY>0)
+					{
+						double angle = std::atan2(angleY, angleX);
+						return carve::geom::VECTOR(std::cos(angle), std::sin(angle), 0.);
+
+					}
+					// Only works if correctly rotation around y axis is in convertIfcCurveSegment 
+					else if (polynomialConstantCntX>0 && polynomialConstantCntZ>0)
+					{
+						double angle = std::atan2(angleZ, angleX);
+						return carve::geom::VECTOR(std::cos(angle), 0., -std::sin(angle));
+					}
+					// Only works if correctly rotation around x axis is in convertIfcCurveSegment 
+					else if (polynomialConstantCntY>0 && polynomialConstantCntZ>0)
+					{
+						double angle = std::atan2(angleZ, angleY);
+						return carve::geom::VECTOR(0., std::cos(angle), std::sin(angle));
+					}
+
+					//TODO: angle in 3D. At first implement additional code in convertIfcCurveSegment
+					/*else if (polynomialConstantCntX>0 && polynomialConstantCntY>0 && polynomialConstantCntZ>0)
+					{
+				
+					}*/			
+				}
+
+#endif
+
+#if defined(OIP_MODULE_EARLYBINDING_IFC4X3_RC4)
+				/*! \brief Calculates a trimming point on the polynomial curve.
+				* \param[in] polynomialCurve	    A pointer to data from \c IfcPolynomialCurve.
+				* \param[in] parameter				A pointer to data from \c IfcCurveSegment.
+				* \return							The location of the trimming point.
+				* \note
+				*/
+				template <>
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcPolynomialCurve>& polynomialCurve,
+					const typename IfcEntityTypesT::IfcParameterValue& parameter) const noexcept(false)
+				{
+					return getPointOnCurve(polynomialCurve, parameter * this->UnitConvert()->getLengthInMeterFactor());
+				}
+				template <>
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcPolynomialCurve>& polynomialCurve,
+					const typename IfcEntityTypesT::IfcNonNegativeLengthMeasure& parameter) const noexcept(false)
+				{
+					return getPointOnCurve(polynomialCurve, parameter * this->UnitConvert()->getLengthInMeterFactor());
+				}
+				
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcPolynomialCurve>& polynomialCurve,
+					const double& parameter) const noexcept(false)
+
+				{
+
+					std::vector<typename IfcEntityTypesT::IfcReal> coefficientX, coefficientY, coefficientZ;

```

Comment:
it would make more sense to use plural here to make it clear that these variables include multiple coefficients

Reply:
Resolved: [98adeb1](https://github.com/tumcms/Open-Infra-Platform/pull/539/commits/98adeb1aa6baac4fd3b81995623cb7bff780ddb0)